File: 240418/G or H 3/G-or-H-3.py
N, K = map(int, input().split())
people = []
for _ in range(N):
    pos, alpha = input().split()
    pos = int(pos) - 1
    people.append([pos, alpha])

line = [0] * (max(list(zip(*people))[0]) + 1)
for pos, alpha in people:
    if alpha == 'G':
        line[pos] = 1
    else:
        line[pos] = 2

max_sum = 0
for i in range(len(line) - K):
    max_sum = max(max_sum, sum(line[i:i + K + 1]))

print(max_sum)

File: 260219/G or H 3/G-or-H-3.cpp
#include <bits/stdc++.h>
using namespace std;


int main() 
{
    int N, K;
    cin >> N >> K;

    const int MAX_POS = 10000;
    vector<int> line(MAX_POS + 1, 0);

    for (int i = 0; i < N; i++) 
    {
        int pos;
        char alpha;
        cin >> pos >> alpha;
        line[pos] = (alpha == 'G') ? 1 : 2;
    }

    int windowSize = K + 1;

    long long windowSum = 0;
    for (int i = 0; i < windowSize; i++) windowSum += line[i];

    long long maxSum = windowSum;

    for (int i = windowSize; i <= MAX_POS; i++) 
    {
        windowSum += line[i];
        windowSum -= line[i - windowSize];
        maxSum = max(maxSum, windowSum);
    }

    cout << maxSum << "\n";
    return 0;
}